src/LaunchDarkly.ServerSdk/Internal/BigSegments/BigSegmentsInternalTypes.cs

```diff
@@ -23,10 +23,16 @@ internal static class BigSegmentsInternalTypes
             string.Format("{0}.g{1}", s.Key, s.Generation.Value);
 
         // This type is used when Evaluator is querying Big Segments
-        internal struct BigSegmentsQueryResult
+        internal readonly struct BigSegmentsQueryResult
```

Comment:
Enforces immutability so the compiler knows it doesn't have to do a defensive copy.